packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -0,0 +1,247 @@
+ * Defines HTTP gateway route path or prefix request match.
+ */
+export abstract class HttpGatewayRoutePathMatch {
+  /**
+   * The value of the path must match the specified value exactly.
+   *
+   * @param pathMatch The exact path to match on
+   */
+  public static exact(pathMatch: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePathSpecificMatch({ exact: pathMatch }, exactPathRewrite);
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   * @param regexMatch The regex used to match the path.
+   */
+  public static regex(regexMatch: string, exactPathRewrite?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePathSpecificMatch({ regex: regexMatch }, exactPathRewrite);
+  }
+
+  /**
+   * The value of the path must match the specified prefix.
+   *
+   * @param prefixMatch This parameter must always start with /, which by itself matches all requests to the virtual service name.
+   *  You can also match for path-based routing of requests. For example, if your virtual service name is my-service.local
+   *  and you want the gateway route to match requests to my-service.local/metrics, your prefix should be /metrics.
+   */
+  public static startingWith(prefixMatch: string, prefixRewrite?: HttpGatewayRoutePrefixPathRewrite): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePrefixMatchImpl(prefixMatch, prefixRewrite);
+  }
+
+  /**
+   * Returns the gateway route path match configuration.
+   */
+  public abstract bind(scope: Construct): HttpGatewayRoutePathMatchConfig;
+}
+
+class HttpGatewayRoutePrefixMatchImpl extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly prefixPathMatch: string,
+    private readonly prefixPathRewrite?: HttpGatewayRoutePrefixPathRewrite,
+  ) {
+    super();
+  }
+
+  bind(scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      prefixPathMatch: this.prefixPathMatch,
+      prefixPathRewrite: this.prefixPathRewrite?.bind(scope).prefixPathPath,
+    };
+  }
+}
+
+class HttpGatewayRoutePathSpecificMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly match: CfnGatewayRoute.HttpPathMatchProperty,
+    private readonly exactPathRewrite?: string,
+  ) {
+    super();
+  }
+
+  bind(_scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      wholePathMatch: this.match,
+      wholePathRewrite: {
+        exact: this.exactPathRewrite,
+      },
+    };
+  }
+}
+
+/**
+ * Prefix properties for HTTP gateway route rewrite.
+ */
+export interface HttpGatewayRoutePrefixPathRewriteConfig {
+  /**
+   * GatewayRoute CFN configuration for HTTP gateway route prefix rewrite.
+   *
+   * @default - rewrite prefix to '/'.
+   */
+  readonly prefixPathPath?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Used to generate HTTP gateway route rewrite other than the host name.
+ */
+export abstract class HttpGatewayRoutePrefixPathRewrite {
```

Comment:
In my opinion, this class is not needed.

I would remove it, and change `HttpGatewayRoutePathMatch.startingWith()` to take a string as the second parameter.

If you want, you can special-case providing that parameter as `/`, and set the `DefaultPrefix` property to `ENABLED` in the CloudFormation template. But I don't like having this class in the API.

Reply:
I still think this class is needed to provide `disableDefaultPrefix` option.

If the second parameter only accepts a `string`, would it just provide an option to:
 - When `string` is provided, this is equivalent to specifying `customPrefix`
 - When `string` is not provided, this is equivalent to not specifying the rewrite an thus `defaultPrefix` is `ENABLED`

?

Reply:
Can you explain to me one thing? In your current API (in this revision of the PR), what is the difference between:

```ts
match: HttpGatewayRoutePathMatch.startingWith('/'),
```

and 

```ts
match: HttpGatewayRoutePathMatch.startingWith('/', HttpGatewayRoutePrefixPathRewrite.customPrefix('/')),
```

?

Reply:
Those two would result in same.

Reply:
Thanks. One more question.

I understand that the default rewrite for the `startingWith()` path match is to rewrite the entire path to `/`.

What about the other `HttpGatewayRoutePathMatch` options? For example, what is the default behavior for:

```ts
match: HttpGatewayRoutePathMatch.exactly('/mypath'),
```

? Is it also to re-write `/mypath` to `/`, or is it something else?

Reply:
> Is it also to re-write `/mypath` to `/`, or is it something else?

Yes, as a default, it is re-written to `/`.

Reply:
Then why did you model rewrites with a union-like class only for `HttpGatewayRoutePathMatch.startingWith()`? Why not do it for `HttpGatewayRoutePathMatchexactly()` and `HttpGatewayRoutePathMatch.regext()` too?